donut/modules/groups/helpers.py

```diff
@@ -40,3 +40,41 @@ def get_group_list_data(fields=None, attrs={}):
     # Return the rows in the form of a list of dicts
     result = [{f: t for f, t in zip(fields, res)} for res in result]
     return result
+
+
+def get_group_data(group_id, fields=None):
+    """
+    Queries the databse and returns member data for the specified group_id.
+
+    Arguments:
+        group_id: The group to look up
+        fields:   The fields to return. If None are specified, then
+                  default_fields are used
+    
+    Returns:
+        result:   The fields and corresponding values of group with group_id.
+                  In the form of a dict with key:value of columnname:columnalue
+    """
+    all_returnable_fields = [
+        "group_id", "group_name", "group_desc", "type", "anyone_can_send",
+        "members_can_send", "newsgroups", "visible", "admin_control_members"
+    ]
+    default_fields = ["group_id", "group_name", "group_desc", "type"]
+    if fields == None:
+        fields = default_fields
+    else:
+        if any(f not in all_returnable_fields for f in fields):
+            return "Invalid field"
+
+    # Build the SELECT and FROM clauses
+    s = sqlalchemy.sql.select(fields).select_from(sqlalchemy.text("groups"))
+
+    # Build the WHERE clause
+    s = s.where(sqlalchemy.text("group_id = :g"))
+
+    # Execute the query
+    result = flask.g.db.execute(s, g=group_id).first()
+
```

Comment:
`result` can be `None` if there is no group with the group id.
We should return `{}` in this case.

Reply:
Done